ui/selectors/selectors.js

```diff
@@ -750,7 +750,10 @@ export function getIpfsGateway(state) {
 }
 
 export function getInfuraBlocked(state) {
-  return Boolean(state.metamask.infuraBlocked);
+  return (
+    state.metamask.networksMetadata[getSelectedNetworkClientId(state)].status ===
+    NetworkStatus.Blocked
```

Comment:
alternatively, this can check if any network is blocked rather than only checking the selected network

Reply:
Having this trigger on any network being blocked by Infura could be breaking as you wouldn't be able to resolve the issue by switching networks since we would be checking all network statuses and you also would not be able to remove the blocked network as it would be a built-in Infura network. Let's leave this as is